### PR TITLE
Remove entry file modification/index.js generation from `expo prebuild`

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- Possible since https://github.com/expo/expo/pull/21643
- Arbitrary entry files in development only work when using `expo-dev-client` or `.expo/.virtual-metro-entry` (SDK +49).

# Test Plan

- TBD 

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/workflow/prebuild.mdx [hunk 1]
```diff
@@ -91,10 +91,10 @@ As of SDK 46, `npx expo prebuild` performs several side effects outside of gener
 In addition to generating the native folders, prebuild also makes the following modifications:
 
 - Generates a **metro.config.js** file (if it doesn't exist).
-- Generates an **index.js** file (if it doesn't exist).
-- Removes the `main` field in the **package.json**.
 - Modifies the `scripts` field in the **package.json**.
 - Modifies the `dependencies` field in the **package.json**.
+- [SDK 48 and below]: Removes the `main` field in the **package.json**.
+- [SDK 48 and below]: Generates an **index.js** file (if it doesn't exist).
```
Comment (amandeepmittal) [inline docs/pages/workflow/prebuild.mdx:97]:
```suggestion
- For SDK 48 and below, removes the `main` field in the **package.json**.
- For SDK 48 and below, generates an **index.js** file (if it doesn't exist).
```

**Comment by brentvatne** [comment]
if I'm using SDK 48, what will I need to do to migrate to use this in SDK 49? will I need to manually change the main file to `.expo/.virtual-metro-entry`? can you add instructions people need to know to the release notes task? https://linear.app/expo/issue/ENG-8316/release-notes

**Review by brentvatne — APPROVED** [review]
approved, we should follow up with information on how to migrate etc as mentioned in comment above

**Comment by EvanBacon** [comment]
@brentvatne the setup guide is in the versioned metro doc https://github.com/expo/expo/blob/main/docs/pages/versions/unversioned/config/metro.mdx#custom-entry-file

**Pull request merged**